kyo-prelude/shared/src/main/scala/kyo2/kernel/Pending.scala

```diff
@@ -7,15 +7,12 @@ import scala.annotation.tailrec
 import scala.language.implicitConversions
 import scala.util.NotGiven
 
-// TODO Constructor should be private but there's an issue with inlining
-case class <[+A, -S](private val curr: A | Kyo[A, S]) extends AnyVal
+opaque type <[+A, -S] = A | Kyo[A, S]
```

Comment:
This is declared as `opaque type <[+A, -S] >: A = A | Kyo[A, S]` in `kyo-core`. Note that I didn't add the `>: A` back, which provides automatic lifting via the type system. The approach using implicit conversion has better usability because it can avoid unexpected nesting of computations (see `NotGiven` in `lift` bellow).